### PR TITLE
Fix `num_classes` arg in F1 metric

### LINKED ISSUE #5662: F1 score > 1
## 🐛 Bug

F1 is > 1

From comet:

image

Thank you!

Cheers,

Francesco

## Please reproduce using the BoringModel

```python
class MySystem(pl.LightningModule):

    def __init__(self, model: nn.Module, criterion: Callable = nn.CrossEntropyLoss, lr :float = 1e-3):
        super().__init__()
        self.model = model
        self.criterion = criterion()
        self.lr = lr
        self.metrics = nn.ModuleDict({
            'accuracy': Accuracy(),
            'f1': F1()
        })
        # this apis are so bad
        
    def compute_metrics(self, out: Tensor, batch: Union[Tensor, Tensor]) -> Dict[str, Tensor]:
        x, y = batch
        metrics = {}
        for name, metric in self.metrics.items():
            metrics[name] = metric(out, y)
        return metrics

    def basic_step(self, batch: Union[Tensor, Tensor], name_space: str) -> Tensor:
        x, y = batch
        out = self.model(x)
        loss = self.criterion(out, y)
        metrics = self.compute_metrics(out, batch)
        self.log_dict({f'{name_space}_loss': loss, **
                       {f'{name_space}_{k}': v for k, v in metrics.items()}})
        return loss

    def training_step(self, batch: Union[Tensor, Tensor], batch_idx: int) -> Tensor:
        loss = self.basic_step(batch, name_space='train')
        return loss

    def validation_step(self, batch: Union[Tensor, Tensor], batch_idx: int) -> Tensor:
        loss = self.basic_step(batch, name_space='val')
        return loss

    def test_step(self, batch: Union[Tensor, Tensor], batch_idx: int) -> Tensor:
        loss = self.basic_step(batch, name_space='test')
        return loss

    def configure_optimizers(self):
        optimizer = optim.AdamW(self.parameters(), lr=self.lr)
        scheduler = optim.lr_scheduler.ReduceLROnPlateau(optimizer)
        return {
            'optimizer': optimizer,
            'lr_scheduler': scheduler,
            'monitor': 'val_loss'
        }

```

The training is correct as loss and accuracy are consistent, 

```python
...
    # our callbacks
    callbacks = [
        ModelCheckpoint(
            monitor='val_loss',
            dirpath=model_checkpoint_dir,
            filename='{epoch:02d}-{val_loss:.2f}'
        ),
        EarlyStopping(monitor='val_loss', patience=10, verbose=True)
    ]
    # using commet
    logger = CometLogger(
        api_key=secrets['COMET_API_KEY'],
        project_name="test",
        workspace="francescosaveriozuppichini"
    )
    logger.log_hyperparams(params)
    
    system = MySystem(model=cnn, lr=params['lr'])

    trainer = Trainer(gpus=1, min_epochs=params['epochs'],
                      progress_bar_refresh_rate=20, logger=logger,
                      callbacks=callbacks)

    trainer.fit(system, train_dl, val_dl)

    print(trainer.test(test_dataloaders=test_dl))

```

### Environment

* CUDA:
        - GPU:
                - GeForce GTX 1080 Ti
        - available:         True
        - version:           11.0
* Packages:
        - numpy:             1.18.4
        - pyTorch_debug:     True
        - pyTorch_version:   1.7.0+cu110
        - pytorch-lightning: 1.1.5
        - tqdm:              4.46.0
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.8.2
        - version:           #45~20.04.1-Ubuntu SMP Fri Jan 15 11:35:04 UTC 2021

[Edit] Forgot to say thank you :)

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #5662
Removes default for `num_classes` arg in `F1` metric, as not setting it correctly can lead to strange results.
Furthermore make sure `F1` is actually tested and not just `FBeta` metric with `beta=1.0`.
cc: @teddykoker 

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by teddykoker — APPROVED** [review]
Perfect, can we get this in the next release?

**Review by SeanNaren — APPROVED** [review]

File: pytorch_lightning/metrics/classification/f_beta.py [hunk 1]
```diff
@@ -211,7 +211,7 @@ class F1(FBeta):
 
     def __init__(
         self,
-        num_classes: int = 1,
+        num_classes,
```
Comment (Borda) [inline pytorch_lightning/metrics/classification/f_beta.py:214]:
```suggestion
        num_classes: int,
```

File: tests/metrics/classification/test_f_beta.py [hunk 2]
```diff
@@ -106,22 +106,26 @@ def _sk_fbeta_multidim_multiclass(preds, target, average='micro', beta=1.0):
...
 class TestFBeta(MetricTester):
     @pytest.mark.parametrize("ddp", [True, False])
     @pytest.mark.parametrize("dist_sync_on_step", [True, False])
     def test_fbeta(
         self, preds, target, sk_metric, num_classes, multilabel, average, beta, ddp, dist_sync_on_step
     ):
+        if beta == 1.0:
+            metric_class = F1
+        else:
+            metric_class = partial(FBeta, beta=beta)
```
Comment (Borda) [inline tests/metrics/classification/test_f_beta.py:119]:
```suggestion
        metric_class = F1 if beta == 1.0 else partial(FBeta, beta=beta)
```

File: tests/metrics/classification/test_f_beta.py [hunk 3]
```diff
@@ -134,12 +138,16 @@ def test_fbeta(
     def test_fbeta_functional(
         self, preds, target, sk_metric, num_classes, multilabel, average, beta
     ):
+        if beta == 1.0:
+            metric_functional = f1
+        else:
+            metric_functional = partial(fbeta, beta=beta)
```
Comment (Borda) [inline tests/metrics/classification/test_f_beta.py:144]:
```suggestion
        metric_functional = f1 if beta == 1.0 else partial(fbeta, beta=beta)
```

**Review by s-rog — APPROVED** [review]
LGTM, just get Borda's comments!

**Review by Borda — APPROVED** [review]
